src/index.ts

```diff
@@ -24,20 +25,26 @@
 }
 
 async function getPRDetails(): Promise<PRDetails> {
-  const { repository, number } = JSON.parse(
-    readFileSync(process.env.GITHUB_EVENT_PATH || "", "utf8")
-  );
-  const prResponse = await octokit.pulls.get({
-    owner: repository.owner.login,
-    repo: repository.name,
-    pull_number: number,
+  const context = github.context;
+  if (!context.payload.pull_request) {
+    throw new Error("This action can only be run on pull requests");
+  }
+
+  const { owner, repo } = context.repo;
+  const pull_number = context.payload.pull_request.number;
+
+  const { data: pullRequest } = await octokit.rest.pulls.get({
+    owner,
+    repo,
+    pull_number,
   });
+
```

Comment:
The `createReviewComment` function uses the `event: 'COMMENT'` parameter, which is appropriate for creating review comments. However, consider whether this should be 'REQUEST_CHANGES' or 'APPROVE' based on the analysis results.